internal/api/handlers/rooms_handler.go

```diff
@@ -0,0 +1,99 @@
+//
+// The above copyright notice and this permission notice shall be included in all
+// copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+
+package handlers
+
+import (
+	"context"
+	"errors"
+	"time"
+
+	portsErrors "github.com/topfreegames/maestro/internal/core/ports/errors"
+
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+
+	"github.com/topfreegames/maestro/internal/core/entities/game_room"
+	"github.com/topfreegames/maestro/internal/core/services/room_manager"
+
+	api "github.com/topfreegames/maestro/pkg/api/v1"
+)
+
+type RoomsHandler struct {
+	roomManager *room_manager.RoomManager
+	api.UnimplementedRoomsServiceServer
+}
+
+func ProvideRoomsHandler(roomManager *room_manager.RoomManager) *RoomsHandler {
+	return &RoomsHandler{
+		roomManager: roomManager,
+	}
+}
+
+func (h *RoomsHandler) UpdateRoomWithPing(ctx context.Context, message *api.UpdateRoomWithPingRequest) (*api.UpdateRoomWithPingResponse, error) {
+	gameRoom, err := h.fromApiUpdateRoomRequestToEntity(message)
+	if err != nil {
+		return nil, status.Error(codes.InvalidArgument, err.Error())
+	}
+	err = h.roomManager.UpdateRoom(ctx, gameRoom)
+	if err != nil {
+		if errors.Is(err, portsErrors.ErrNotFound) {
+			return nil, status.Error(codes.NotFound, err.Error())
+		}
+		return nil, status.Error(codes.Unknown, err.Error())
+	}
+	return &api.UpdateRoomWithPingResponse{Success: true}, nil
+}
+
+func (h *RoomsHandler) fromApiUpdateRoomRequestToEntity(request *api.UpdateRoomWithPingRequest) (*game_room.GameRoom, error) {
+	status, err := fromStringToGameRoomStatus(request.GetStatus())
+	if err != nil {
+		return nil, err
+	}
+
+	return &game_room.GameRoom{
+		ID:          request.GetRoomName(),
+		SchedulerID: request.GetSchedulerName(),
+		Status:      status,
+		Metadata:    request.Metadata.AsMap(),
+		LastPingAt:  time.Unix(request.GetTimestamp(), 0),
+	}, nil
+}
+
+func fromStringToGameRoomStatus(value string) (game_room.GameRoomStatus, error) {
+	switch value {
+	case "pending":
+		return game_room.GameStatusPending, nil
+	case "unready":
+		return game_room.GameStatusUnready, nil
+	case "ready":
+		return game_room.GameStatusReady, nil
+	case "occupied":
+		return game_room.GameStatusOccupied, nil
+	case "terminating":
+		return game_room.GameStatusTerminating, nil
+	case "error":
+		return game_room.GameStatusError, nil
+	default:
+		return game_room.GameStatusPending, portsErrors.NewErrInvalidArgument("invalid value for GameRoomStatus: %s", value)
+	}
```

Comment:
We have a "[to string" method](https://github.com/topfreegames/maestro/blob/next/internal/core/entities/game_room/game_room.go#L47-L64) for the status enum, what do you think to place this "from string" in the same place?

Reply:
Good ideia

Reply:
f7d5d2b